docs/tls/README.md

```diff
@@ -1,90 +1,145 @@
+* A certificate signed by a CA contains information about the issued identity (e.g. name, expiry, public key) and any intermediate certificates. The root CA is not included.
 
-Before generating your self-signed certificate, note that
+## <a name="generate-use-self-signed-keys-certificates"></a>3. Generate and use Self-signed Keys and Certificates with Minio
 
-- Minio supports only key/certificate in PEM format on Linux.
-- Minio supports only key/certificate in PEM format on Windows. We don't support PFX certificates currently.
+This section describes how to generate a self-signed certificate using various tools:
 
-### Using generate_cert.go
+3.1 [Use generate_cert.go to Generate a Certificate](#using-go) 
+3.2 [Use OpenSSL to Generate a Certificate](#using-open-ssl) 
+3.3 [Use OpenSSL (with IP address) to Generate a Certificate](#using-open-ssl-with-ip) 
+3.4 [Use GnuTLS (for Windows) to Generate a Certificate](#using-gnu-tls)
 
-Download [generate_cert.go](https://golang.org/src/crypto/tls/generate_cert.go?m=text). This is a simple go tool to generate self-signed certificates.
+**Note:**
+* Minio only supports keys and certificates in PEM format on Linux and Windows.
+* Minio doesn't currently support PFX certificates.
 
-`generate_cert.go` already provides SAN certificates with DNS and IP entries:
+### <a name="using-go"></a>3.1 Use generate_cert.go to Generate a Certificate
+
+Download [`generate_cert.go`](https://golang.org/src/crypto/tls/generate_cert.go?m=text).
+
+`generate_cert.go` is a simple *Go* tool to generate self-signed certificates, and provides SAN certificates with DNS and IP entries:
 
 ```sh
 go run generate_cert.go -ca --host "10.10.0.3"
 ```
 
-### Using OpenSSL
+A response similar to this one should be displayed:
+
+```
+2018/11/21 10:16:18 wrote cert.pem
+2018/11/21 10:16:18 wrote key.pem
+```
+
+### <a name="using-open-ssl"></a>3.2 Use OpenSSL to Generate a Certificate
+
+Use one of the following methods to generate a certificate using `openssl`:
+
+* 3.2.1 [Generate a private key with ECDSA](#generate-private-key-with-ecdsa) 
+* 3.2.2 [Generate a private key with RSA](#generate-private-key-with-rsa)
+
+#### 3.2.1 <a name="generate-private-key-with-ecdsa"></a>Generate a private key with ECDSA.
 
-**Generate the private key**:
+Use the following command to generate a private key with ECDSA:
 
-1. **ECDSA:**  
 ```sh
 openssl ecparam -genkey -name prime256v1 | openssl ec -out private.key
 ```
-or protect the private key additionally with a password:  
+
+A response similar to this one should be displayed:
+
+```
+read EC key
+writing EC key
+```
+
+Alternatively, use the following command to generate a private ECDSA key protected by a password:
+
 ```sh
 openssl ecparam -genkey -name prime256v1 | openssl ec -aes256 -out private.key -passout pass:PASSWORD
 ```
 
-Notice that the NIST curves P-384 and P-521 are not supported yet.
+**Note:** NIST curves P-384 and P-521 are not currently supported.
```

Comment:
typo: s / not currently / currently not

Reply:
@aead This is actually correct English.

Reply:
Oh - sry @jamesneimanconsulting - never mind.